account_statement_completion/models/completion_rules.py

```diff
@@ -37,96 +61,55 @@ class StatementCompletionRule(models.Model):
+                    eval_dict,
+                    mode="exec",
+                    nocopy=True
+                )
+                if eval_dict.get("result"):
+                    break
         return dict()
 
-    def get_from_amount(self, stmts_vals, st_line):
-        """ If line amount match an open supplier invoice,
-            update partner and account. """
-        amount = st_line['amount']
-        res = {}
-        # We check only for debit entries
-        if amount < 0:
-            invoice_obj = self.env['account.move']
-            invoices = invoice_obj.search(
-                [('move_type', '=', 'in_invoice'), ('state', '=', 'posted'),
-                 ('amount_total', '=', abs(amount))])
-            res = {}
-            if invoices:
-                if len(invoices) == 1:
-                    partner = invoices.partner_id
-                    res['partner_id'] = partner.commercial_partner_id.id
-                else:
-                    partner = invoices[0].partner_id
-                    for invoice in invoices:
-                        if invoice.partner_id.id != partner.id:
-                            logger.warning(
-                                f"Line named {st_line['name']} (Ref:{st_line['ref']})"
-                                f" was matched by "
-                                'more than one invoice while looking on open'
-                                ' supplier invoices')
-                    res['partner_id'] = partner.commercial_partner_id.id
-        return res
-
-    def get_from_move_line_ref(self, stmts_vals, st_line):
-        ''' Update partner if same reference is found '''
-        ref = st_line.get('ref')
-        res = dict()
-        if not ref:
-            return res
-        partner = None
-
-        # Search move lines
-        move_line_obj = self.env['account.move.line']
-        move_lines = move_line_obj.search(
-            [('ref', '=', ref), ('partner_id', '!=', None)])
-        if move_lines:
-            partner = move_lines[0].partner_id
-
-        if partner:
-            res['partner_id'] = partner.commercial_partner_id.id
-
-        return res
-
-    def get_from_payment_line(self, stmt_vals, st_line):
-        """ Search in account.payment.line """
-        ref = st_line.get('ref')
-        res = dict()
-        if not ref:
-            return res
-
-        payment_line = self.env['bank.payment.line'].search([
-            ('name', '=', ref)
-        ], limit=1, order='date desc')
-        if payment_line:
-            res['partner_id'] = payment_line.partner_id.\
-                commercial_partner_id.id
-        return res
+    def _get_base_dict(self, stmts_vals, stmt_line):
+        """ Return the values usable in the code template """
+        return {
+            "result": {},
+            "stmts_vals": stmts_vals,
+            "stmt_line": stmt_line,
+            "line_amount": int(stmt_line["amount"]),
+            "ref": stmt_line.get('ref'),
```

Comment:
```suggestion
```

Reply:
Same as the env variable, to me giving this access isn't a bad idea.